mxcube3/ui/components/SampleView/ContextMenu.js

```diff
@@ -40,13 +40,22 @@ export default class ContextMenu extends React.Component {
 
   showModal(modalName) {
     const { sampleID, defaultParameters, shape, sampleData } = this.props;
+    let typePrefix = '';
+    if (modalName === 'Characterisation') {
+      typePrefix = 'c-';
+    } else if (modalName === 'DataCollection') {
+      typePrefix = 'd-';
+    } else if (modalName === 'Helical') {
+      typePrefix = 'l-';
+    }
+
     this.props.showForm(
       modalName,
       [sampleID],
       { parameters:
         { path: sampleData.sampleName,
           ...defaultParameters[modalName.toLowerCase()],
-          prefix: sampleData.defaultPrefix
```

Comment:
A small observation here, we should be care full when we are dealing with prefixes, run numbers and image numbers to make sure that they are as configurable as possible. Its also good if we can avoid logic related to those on the client as far as possible.

So my suggestion would be to implement this on the server logic.